MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffFittingPresenter.cpp

```diff
@@ -0,0 +1,1126 @@
+
+      // delete previous bank added to the list
+      m_view->clearFittingListWidget();
+
+      for (size_t i = 0; i < runNumVector.size(); i++) {
+
+        // get the last split in vector which will be bank
+        std::string currentRun = (runNumVector[i]);
+
+        m_view->addRunNoItem(currentRun);
+      }
+
+      if (multiRun) {
+        m_view->enableFittingListWidget(true);
+
+        auto currentIndex = m_view->getFittingListWidgetCurrentRow();
+        if (currentIndex == -1)
+          m_view->setFittingListWidgetCurrentRow(0);
+      } else {
+        m_view->enableFittingListWidget(false);
+      }
+    }
+
+    else {
+      // upon invalid file
+      // disable the widgets when only one related file found
+      m_view->enableFittingListWidget(false);
+
+      m_view->clearFittingListWidget();
+    }
+
+  } catch (std::runtime_error &re) {
+    m_view->userWarning("Unable to insert items: ",
+                        "Could not add list widget; " +
+                            static_cast<std::string>(re.what()) +
+                            ". Please try again");
+  }
+}
+
+void EnggDiffFittingPresenter::setDefaultBank(
+    const std::vector<std::string> &splittedBaseName,
+    const std::string &selectedFile) {
+
+  if (!splittedBaseName.empty()) {
+
+    std::string bankID = (splittedBaseName.back());
+    auto combo_data = m_view->getFittingComboIdx(bankID);
+
+    if (combo_data > -1) {
+      m_view->setBankIdComboBox(combo_data);
+    } else {
+      m_view->setFittingRunNo(selectedFile);
+    }
+  }
+  // check if the vector is not empty so that the first directory
+  // can be assigned to text-field when number is given
+  else if (!m_view->getFittingRunNumVec().empty()) {
+    auto firstDir = m_view->getFittingRunNumVec().at(0);
+    auto intialDir = firstDir;
+    m_view->setFittingRunNo(intialDir);
+  }
+  // if nothing found related to text-field input
+  else if (!m_view->getFittingRunNo().empty())
+    m_view->setFittingRunNo(selectedFile);
+}
+
+bool EnggDiffFittingPresenter::isDigit(std::string text) {
+  for (size_t i = 0; i < text.size(); i++) {
+    char *str = &text[i];
+    if (std::isdigit(*str)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+void EnggDiffFittingPresenter::plotFitPeaksCurves() {
+  AnalysisDataServiceImpl &ADS = Mantid::API::AnalysisDataService::Instance();
+  std::string singlePeaksWs = "engggui_fitting_single_peaks";
+
+  if (!ADS.doesExist(singlePeaksWs) && !ADS.doesExist(g_focusedFittingWSName)) {
+    g_log.error() << "Fitting results could not be plotted as there is no " +
+                         singlePeaksWs + " or " + g_focusedFittingWSName +
+                         " workspace found.\n";
+    m_view->showStatus("Error while fitting peaks");
+    return;
+  }
+
+  try {
+    auto focusedPeaksWS =
+        ADS.retrieveWS<MatrixWorkspace>(g_focusedFittingWSName);
+    auto focusedData = ALCHelper::curveDataFromWs(focusedPeaksWS);
```

Comment:
I know this has not been introduced in this PR, but just wanted to say that, although this helper was initially written for the muon ALC interface, we should rename it and move it to a shared location if it is going to be useful for other interfaces.